### PR TITLE
[APB-3199][SM, MP, AO]: Move expiry trigger to when invitations are found rather than accepted

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientauthorisation/service/InvitationsStatusUpdateScheduler.scala [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    })
+    actorSystem.scheduler.scheduleOnce(
+      5.seconds,
+      taskActor,
+      "<start invitation status update scheduler>"
+    )
+  } else
+    Logger(getClass).warn("invitation status update scheduler is not enabled.")
+
+  private def updateInvitations(): Future[Unit] =
```
Comment (arturopala) [inline app/uk/gov/hmrc/agentclientauthorisation/service/InvitationsStatusUpdateScheduler.scala]: I think this should be implemented in a separate service injected

Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentclientauthorisation/service/InvitationsStatusUpdateScheduler.scala]: why do we need to create a separate service for this ? i think creating a service class may not be correct as this wont be servicing any requests

Comment (arturopala) [reply app/uk/gov/hmrc/agentclientauthorisation/service/InvitationsStatusUpdateScheduler.scala]: in principle scheduler logic should be separated from business logic, we should be able to test or use this feature outside scheduled job context

Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentclientauthorisation/service/InvitationsStatusUpdateScheduler.scala]: yeah, but here, the `updateInvitations` logic doesn't make much sense out side of scheduler -- in future if we need to re-use then we can actually move the logic out of the scheduler
I think its an overkill to create a service for this small logic but i can do -

File: app/uk/gov/hmrc/agentclientauthorisation/service/InvitationsService.scala [hunk 2]
```diff
@@ -209,6 +209,19 @@ class InvitationsService @Inject()(
       invitationsRepository.findInvitationInfoBy(arn, service, clientId, status, createdOnOrAfter)
     }
 
+  def findAndUpdateExpiredInvitations()(implicit ec: ExecutionContext): Future[Unit] =
+    monitor(s"Repository-Find-And-Update-Expired-Invitations") {
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentclientauthorisation/service/InvitationsService.scala]: not sure if we can give any better name for this method

**Pull request merged**